lib/app/service/ntp_service.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:app/app/service/service.dart';
+import 'package:injectable/injectable.dart';
+import 'package:logging/logging.dart';
+import 'package:ntp/ntp.dart';
+import 'package:rxdart/rxdart.dart';
+
+/// Maximum time to wait for NTP server response
+const _updateTimeout = Duration(seconds: 10);
+
+/// Period of time between NTP server requests
+const _updatePeriod = Duration(minutes: 1);
+
+/// Delay between app resume and NTP server request
+const _resumedDelay = Duration(milliseconds: 500);
+
+Timer? _updateTimer;
+
+/// Service that helps watching app lifecycle state.
+@singleton
+class NtpService {
+  NtpService(this.appLifecycleService);
+
+  final AppLifecycleService appLifecycleService;
+
+  final _log = Logger('NtpService');
+
+  bool _periodicUpdatesEnabled = false;
+
+  Future<void> init() async {
+    /// Start periodic updates, but don't wait for it to complete
+    /// because it may take a long time (network request of the
+    /// first update)
+    unawaited(_startPeriodicUpdates());
+
+    appLifecycleService.appLifecycleStateStream
+        .where((state) => state == AppLifecycleState.resumed)
+        .delay(_resumedDelay)
+        .listen(
+      (event) {
+        _log.info('App resumed, updating offset...');
+        update();
+      },
+    );
+  }
+
+  @disposeMethod
+  void dispose() {
+    _stopPeriodicUpdates();
```

Comment:
For singleton's this is not necessary

Reply:
In this application it's not necessary, but I thought about separated package for NtpService.

Reply:
So, you **can** provide method to dispose a singleton. Why not do it if you are not sure how it will be used?
https://pub.dev/packages/injectable#disposing-of-singletons